astropy/coordinates/builtin_frames/skyoffset.py

```diff
@@ -43,36 +43,20 @@ def make_skyoffset_cls(framecls):
     if framecls in _skyoffset_cache:
         return _skyoffset_cache[framecls]
 
-    # the class of a class object is the metaclass
-    framemeta = framecls.__class__
-
-    class SkyOffsetMeta(framemeta):
-        """
-        This metaclass renames the class to be "SkyOffset<framecls>" and also
-        adjusts the frame specific representation info so that spherical names
-        are always "lon" and "lat" (instead of e.g. "ra" and "dec").
-        """
-
-        def __new__(cls, name, bases, members):
-            # Only 'origin' is needed here, to set the origin frame properly.
-            members['origin'] = CoordinateAttribute(frame=framecls, default=None)
-
-            # This has to be done because FrameMeta will set these attributes
-            # to the defaults from BaseCoordinateFrame when it creates the base
-            # SkyOffsetFrame class initially.
-            members['_default_representation'] = framecls._default_representation
-            members['_default_differential'] = framecls._default_differential
-
-            newname = name[:-5] if name.endswith('Frame') else name
```

Comment:
This was, I think, trying too hard to even account for generic subclasses of `SkyOffsetFrame`. It is not actually useful since below a subclass of `SkyOffsetFrame` is created...

Reply:
Okay, I didn't know about this, and seeing it now kind of answers my previous question about what the problem was.  But if you remove this metaclass are the previous changes needed?  Or do you think the previous changes are still good to have in case another case like this metaclass arises?

Reply:
Yes, the changes in `baseframe` are still needed - I did this before the other ones. It seems a tricky business merging the attributes, keeping order, and possibly overriding ones...